src/main/java/org/fusesource/jansi/io/AnsiOutputStream.java

```diff
@@ -85,7 +85,7 @@ public int getTerminalWidth() {
     private final Charset cs;
 
     private final WidthSupplier width;
-    private final AnsiProcessor processor;
+    private AnsiProcessor processor;
```

Comment:
You should keep the processor final and pass it to the constructor instead.

Reply:
@gnodet I tried that, but the test won't pass in that case, because there are two processors: 'ap' and 'processor', but the AnsiOutputstream constructor only allows to set the 'processor' variable and not the 'ap' variable.
When setting both processors it'll work.
I could add the 'ap' processor to the AnsiOutputstreams constructor if you want.

Reply:
No change should be needed.  You need to pass the correct AnsiMode. Default in the constructor and you should hardcode it in the HtmlAnsiOutputStream constructor.

Reply:
Ok imma try that out

Reply:
@gnodet Ok just tried it out.
I passed over each AnsiMode (Default, Strip and Force) and none of them worked, when the 'ap' processor is not set.

Reply:
Not sure what you're doing.  The `AnsiOuputStream` constructor calls `setMode(mode)` and if you pass `AnsiMode.Default`, the `setMode` should end up doing `ap = processor`...

Reply:
@gnodet Yeah, but this:
```java
public void setMode(AnsiMode mode) {
        ap= mode == AnsiMode.Strip ? new AnsiProcessor(out) : mode == AnsiMode.Force || processor == null ? new ColorsAnsiProcessor(out, colors) : processor;
        this.mode = mode;
    }
```
generates a `new AnsiProcessor` even though I need the new `AnsiToHtmlProcessor`.

Reply:
and why would you pass `AnsiMode.Strip` ?  Your HtmlAnsiOutputStream should not have the `mode` parameter in its constructor and pass `AnsiMode.Default` when calling `super(xxx)`.

Reply:
@gnodet Oh ok It took me 5 minutes to decipher that 'if' statement, and you're right, it should set `ap = processor` if its AnsiMode.Default.
Idk why thats not working...

Reply:
@gnodet 
![image](https://user-images.githubusercontent.com/59899645/122392592-48d4e580-cf74-11eb-980c-721c8f85dc6c.png)
Where is that AnsiMode.Strip???

Reply:
@gnodet Ok managed to make it work and removed the setter methods.